### PR TITLE
xmerl_xsd: add some spec for exported functions

### PR DESCRIPTION
**Pull request by FieldsOfGold06**

### CODE REVIEW

File: lib/xmerl/src/xmerl_xsd.erl [hunk 1]
```diff
@@ -281,6 +288,14 @@ process_validate(Schema,Xml)
 
 Equivalent to [process_validate(Schema, Xml, [])](`process_validate/3`).
 """.
+-spec process_validate(Schema,Element,Options) -> Result when
+      Schema   :: string(),
+      Element :: #xmlElement{},
```
Comment (bmk) [inline lib/xmerl/src/xmerl_xsd.erl:293]: Element :: xmerl:element().

File: lib/xmerl/src/xmerl_xsd.erl [hunk 2]
```diff
@@ -281,6 +288,14 @@ process_validate(Schema,Xml)
...
+      Options  :: option_list(),
+      Result   :: {#xmlElement{},State} | {error,Reasons},
```
Comment (bmk) [inline lib/xmerl/src/xmerl_xsd.erl:295]: Result :: {xmerl:element(), State} | {error, Reason}

File: lib/xmerl/src/xmerl_xsd.erl [hunk 3]
```diff
@@ -116,6 +116,13 @@ validate(Xml,State)
 
 Equivalent to [validate(Element, State, [])](`validate/3`).
 """.
+-spec validate(Element,State) -> Result when
+      Element :: #xmlElement{},
```
Comment (bmk) [inline lib/xmerl/src/xmerl_xsd.erl:120]: Element :: xmerl:element()

File: lib/xmerl/src/xmerl_xsd.erl [hunk 4]
```diff
@@ -116,6 +116,13 @@ validate(Xml,State)
...
+      State :: global_state(),
+      Result :: {ValidElement,global_state()} | {error,Reasons},
+      ValidElement :: #xmlElement{},
```
Comment (bmk) [inline lib/xmerl/src/xmerl_xsd.erl:123]: ValidElement :: xmerl:element()

**Review by lthor — COMMENTED** [review]
Have same comments as Micke.

File: lib/xmerl/src/xmerl_xsd.erl (hunk 1 above)
Comment (FieldsOfGold06) [reply lib/xmerl/src/xmerl_xsd.erl]: fixed

File: lib/xmerl/src/xmerl_xsd.erl (hunk 2 above)
Comment (FieldsOfGold06) [reply lib/xmerl/src/xmerl_xsd.erl]: fixed

File: lib/xmerl/src/xmerl_xsd.erl (hunk 3 above)
Comment (FieldsOfGold06) [reply lib/xmerl/src/xmerl_xsd.erl]: fixed

File: lib/xmerl/src/xmerl_xsd.erl (hunk 4 above)
Comment (FieldsOfGold06) [reply lib/xmerl/src/xmerl_xsd.erl]: fixed

**Review by bmk — APPROVED** [review]

**Pull request merged**